### PR TITLE
PR from vacation branch 

### PR DESCRIPTION
**Pull request by MurodjonMukhamadaliev**
CRUD completed and tested

### CODE REVIEW

File: src/main/java/com/exadel/coolDesking/vacation/VacationController.java [hunk 1]
```diff
@@ -1,14 +1,44 @@
...
 
 @RequiredArgsConstructor
-@RequestMapping(path = "/vacation")
+@RequestMapping(value = "/vacation")
 @RestController
 public class VacationController {
 
     private final VacationService vacationService;
+
+    @PostMapping("/create")
```
Comment (gargatuma) [inline src/main/java/com/exadel/coolDesking/vacation/VacationController.java:18]: No need for '/create', just use PostMapping

File: src/main/java/com/exadel/coolDesking/vacation/VacationController.java [hunk 2]
```diff
@@ -1,14 +1,44 @@
...
+        return ResponseEntity.ok(vacationService.getVacations());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<?> GetVacationById(@PathVariable UUID id) {
+        return ResponseEntity.ok(vacationService.getVacation(id));
+
+    }
+
+    @PutMapping("/edit/{id}")
```
Comment (gargatuma) [inline src/main/java/com/exadel/coolDesking/vacation/VacationController.java:34]: just '/{id}' no need for '/edit'

File: src/main/java/com/exadel/coolDesking/vacation/VacationController.java [hunk 3]
```diff
@@ -1,14 +1,44 @@
...
+    public ResponseEntity<?> edit(@PathVariable UUID id, @RequestBody VacationUpdateDto vacationUpdateDto) {
+        return ResponseEntity.status(HttpStatus.CREATED).body(vacationService.editVacation(vacationUpdateDto, id));
+    }
+
+    @DeleteMapping("/delete/{id}")
```
Comment (gargatuma) [inline src/main/java/com/exadel/coolDesking/vacation/VacationController.java:39]: just '/{id}' no need for '/delete'

File: src/main/java/com/exadel/coolDesking/vacation/VacationCreateDto.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import javax.validation.constraints.NotNull;
+import java.time.LocalDate;
+import java.util.UUID;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+public class VacationCreateDto {
+   // private UUID id;
```
Comment (gargatuma) [inline src/main/java/com/exadel/coolDesking/vacation/VacationCreateDto.java:15]: remove unnecessary comments

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,7 +1,7 @@
 spring.datasource.url=jdbc:postgresql://localhost:5432/hotdesking
 spring.datasource.driver-class-name=org.postgresql.Driver
-spring.datasource.username=aweasadbek
-spring.datasource.password=1234
+spring.datasource.username=postgres
+spring.datasource.password=1199
```
Comment (gargatuma) [inline src/main/resources/application.properties:4]: don't change username and password

File: src/main/java/com/exadel/coolDesking/workspace/WorkplaceController.java [hunk 6]
```diff
@@ -31,6 +31,7 @@
      * @param office_id, workplace_id
      * @return Workplace
      */
+
     @GetMapping("/workplace/{workplace_id}")
```
Comment (gargatuma) [inline src/main/java/com/exadel/coolDesking/workspace/WorkplaceController.java:35]: Why changed this class?

**Review by gargatuma — CHANGES_REQUESTED** [review]

**Review by gargatuma — CHANGES_REQUESTED** [review]

**Review by gargatuma — DISMISSED** [review]

File: src/main/java/com/exadel/coolDesking/vacation/VacationService.java [hunk 7]
```diff
@@ -1,10 +1,65 @@
...
+        return vacationResponseDtos;
+    }
+
+    public VacationResponseDto getVacation(UUID vacationId) {
+        return vacationMapper.entityToResponseDto(vacationRepository.findById(vacationId).orElseThrow(
+                () -> new NotFoundException("Vacation not found", Vacation.class, "vacationId")));
+
+    }
+
+    public UUID createVacation(VacationCreateDto vacationCreateDto) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/vacation/VacationService.java:36]: return dto with the `id` field

**Review by ohalushchak-exadel — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 5 above)
Comment (ohalushchak-exadel) [reply src/main/resources/application.properties:4]: ^ @MurodjonMukhamadaliev

**Review by ohalushchak-exadel — CHANGES_REQUESTED** [review]

**Review by ohalushchak-exadel — APPROVED** [review]

**Review by viktarkolbik — APPROVED** [review]

**Pull request merged**